daml-script/export/src/main/scala/com/daml/script/export/TreeUtils.scala

```diff
@@ -194,12 +194,15 @@ object TreeUtils {
   def treeEventCreatedConsumedCids(
       event: TreeEvent.Kind,
       tree: TransactionTree,
-  ): (Set[ContractId], Set[ContractId]) = {
-    var created = Set.empty[ContractId]
+  ): (Seq[SimpleCreatedContract], Set[ContractId]) = {
+    var created = Seq.empty[SimpleCreatedContract]
     var consumed = Set.empty[ContractId]
     traverseEventInTree(event, tree) {
       case (_, Kind.Created(value)) =>
-        created += ContractId(value.contractId)
+        created :+= SimpleCreatedContract(
```

Comment:
Here is an example of complexity issue. It is not clear what is the complexity of this operation. 

I think, the default implementation of `Seq` is `List`, so your operation is linear in the size of the list.
``` 
 created +:= SimpleCreatedContract
``` 

or 
```suggestion
        created ::= SimpleCreatedContract
```
 
If the order of `created` is relevant, you will have to reverse it before returning it.

Reply:
well spotted, thank you